Userland/Libraries/LibJS/Runtime/SharedArrayBufferPrototype.cpp

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2023, Shannon Booth <shannon.ml.booth@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Function.h>
+#include <LibJS/Runtime/AbstractOperations.h>
+#include <LibJS/Runtime/GlobalObject.h>
+#include <LibJS/Runtime/SharedArrayBufferConstructor.h>
+#include <LibJS/Runtime/SharedArrayBufferPrototype.h>
+
+namespace JS {
+
+SharedArrayBufferPrototype::SharedArrayBufferPrototype(Realm& realm)
+    : PrototypeObject(realm.intrinsics().object_prototype())
+{
+}
+
+ThrowCompletionOr<void> SharedArrayBufferPrototype::initialize(Realm& realm)
+{
+    auto& vm = this->vm();
+
+    MUST_OR_THROW_OOM(Base::initialize(realm));
+    u8 attr = Attribute::Writable | Attribute::Configurable;
+    define_native_function(realm, vm.names.slice, slice, 2, attr);
+    define_native_accessor(realm, vm.names.byteLength, byte_length_getter, {}, Attribute::Configurable);
+
+    // 25.2.4.4 SharedArrayBuffer.prototype [ @@toStringTag ], https://tc39.es/ecma262/#sec-sharedarraybuffer.prototype.toString
+    define_direct_property(vm.well_known_symbol_to_string_tag(), PrimitiveString::create(vm, vm.names.SharedArrayBuffer.as_string()), Attribute::Configurable);
+
+    return {};
+}
+
+// 25.2.4.1 get SharedArrayBuffer.prototype.byteLength, https://tc39.es/ecma262/#sec-get-sharedarraybuffer.prototype.bytelength
+JS_DEFINE_NATIVE_FUNCTION(SharedArrayBufferPrototype::byte_length_getter)
+{
+    // 1. Let O be the this value.
+    // 2. Perform ? RequireInternalSlot(O, [[ArrayBufferData]]).
+    auto array_buffer_object = TRY(typed_this_value(vm));
+
+    // 3. If IsSharedArrayBuffer(O) is true, throw a TypeError exception.
+    // FIXME: Check for shared buffer
+
+    // 4. Let length be O.[[ArrayBufferByteLength]].
+    // 5. Return 𝔽(length).
+    return Value(array_buffer_object->byte_length());
+}
+
+// 25.2.4.3 SharedArrayBuffer.prototype.slice ( start, end ), https://tc39.es/ecma262/#sec-sharedarraybuffer.prototype.toString
+JS_DEFINE_NATIVE_FUNCTION(SharedArrayBufferPrototype::slice)
+{
+    auto& realm = *vm.current_realm();
+
+    // 1. Let O be the this value.
+    // 2. Perform ? RequireInternalSlot(O, [[ArrayBufferData]]).
+    auto array_buffer_object = TRY(typed_this_value(vm));
+
+    // 3. If IsSharedArrayBuffer(O) is false, throw a TypeError exception.
+    // FIXME: Check for shared buffer
+
+    // 4. Let len be O.[[ArrayBufferByteLength]].
+    auto length = array_buffer_object->byte_length();
+
+    // 5. Let relativeStart be ? ToIntegerOrInfinity(start).
+    auto relative_start = TRY(vm.argument(0).to_integer_or_infinity(vm));
```

Comment:
Please `auto start = vm.argument(0);` once at the beginning for readability (same for `end`)

Reply:
Yup, much nicer. Also included a comment for same change on the ArrayBufferPrototype.